File: src/web/magmac/app/lang/BlockFolder.ts
import { DivideState } from "../../../magmac/app/compile/rule/divide/DivideState";
import { Folder } from "../../../magmac/app/compile/rule/fold/Folder";
export class BlockFolder {
	public fold( state : DivideState,  c : char) : DivideState { let appended : var=state.append( c);if(true){ return appended.advance( );;}return appended;;}
	public createDelimiter() : String {return "";;}
}

File: src/java/magmac/app/lang/BlockFolder.java
package magmac.app.lang;

import magmac.app.compile.rule.divide.DivideState;
import magmac.app.compile.rule.fold.Folder;

public class BlockFolder implements Folder {
    @Override
    public DivideState fold(DivideState state, char c) {
        var appended = state.append(c);
        if ('{' == c) {
            return appended.advance();
        }
        return appended;
    }

    @Override
    public String createDelimiter() {
        return "";
    }
}
